op-program/client/interop/consolidate.go

```diff
@@ -203,15 +197,9 @@ func newConsolidateCheckDeps(transitionState *types.TransitionState, chains []et
 	}, nil
 }
 
-func (d *consolidateCheckDeps) Check(
-	chain eth.ChainID,
-	blockNum uint64,
-	timestamp uint64,
-	logIdx uint32,
-	logHash common.Hash,
-) (includedIn supervisortypes.BlockSeal, err error) {
+func (d *consolidateCheckDeps) Contains(chain eth.ChainID, query supervisortypes.ContainsQuery) (includedIn supervisortypes.BlockSeal, err error) {
```

Comment:
This looks like an op-program bug: `Contains` should check all the query parameters. Here it can load by block-number, and the hash etc. can mismatch and would be ignored.

Reply:
cc @Inphi

Reply:
Yup. This is being fixed here - https://github.com/ethereum-optimism/optimism/pull/14101.

Reply:
I did notice that, but since it's using an Oracle, I figure that it *can* make assumptions about the data its getting?

Would the correction here to simply check that the returned data matches the request? Easy to fix if so.

Reply:
Yes, I think so. But it's fine to do so later in the PR of @Inphi imho, it's a simple change.

Reply:
> Would the correction here to simply check that the returned data matches the request? Easy to fix if so.

That's right. The other PR doesn't check all timestamp invariants. Specifically, the message expiry and "ErrFuture" handling. Ideally, the op-supervisor can implement those checks given the result of this function so we don't duplicate the same logic in the op-program.

Reply:
The op-supervisor implementation is different though, since it can produce an `ErrFuture` in case of missing data. Whereas in op-program it should always return `ErrConflict` if e.g. the block number is out of range.

I think reusing the cross-checks from the supervisor, but swapping the implementation of the DB-lookup functions, is the right abstraction to share code.